deployment/deployment.go

```diff
@@ -146,14 +155,22 @@ func CheckTriggererEnabled(client *houston.Client) bool {
 func Create(label, ws, releaseName, cloudRole, executor, airflowVersion, dagDeploymentType, nfsLocation, gitRepoURL, gitRevision, gitBranchName, gitDAGDir, sshKey, knownHosts string, gitSyncInterval, triggererReplicas int, client *houston.Client, out io.Writer) error {
 	vars := map[string]interface{}{"label": label, "workspaceId": ws, "executor": executor, "cloudRole": cloudRole}
 
-	if CheckPreCreateNamespaceDeployment(client) {
+	if CheckPreCreateNamespaceDeployment(client) && !CheckNamespaceFreeformEntryDeployment(client) {
```

Comment:
nit: we can get rid of individual function call to get a field value from appConfig, but rather have a single call to Houston to get all the appConfig and then directly picking the individual field throughout the `Create` method. This will save multiple API calls to Houston to fetch the same appConfig data.

```
appConfig, err := AppConfig(client)
if err != nil {
                 fmt.Println("Error fetching feature flags from Houston: %s", err.Error())
}
if appConfig.Flags.ManualNamespaceNames && !appConfig.Flags.NamespaceFreeformEntry {
		namespace, err := getDeploymentSelectionNamespaces(client, out)
		if err != nil {
			return err
		}
		vars["namespace"] = namespace
}

if appConfig.Flags.ManualNamespaceNames && appConfig.Flags.NamespaceFreeformEntry {
		namespace, err := getDeploymentNamespaceName()
		if err != nil {
			return err
		}
		vars["namespace"] = namespace
}
if releaseName != "" && appConfig.ManualReleaseNames {
		vars["releaseName"] = releaseName
}
.......
if appConfig.Flags.TriggererEnabled {
		vars["triggererReplicas"] = triggererReplicas
}
```

Reply:
Yes I think it does make sense to do.